helper.go

```diff
@@ -463,24 +463,45 @@ func marshalObjectIdentifier(oid []int) (ret []byte, err error) {
 	return
 }
 
-func marshalOID(oid string) ([]byte, error) {
+func oidSplitToIntArray(oid string, sep string) ([]int, error) {
 	var err error
-
-	// Encode the oid
-	oid = strings.Trim(oid, ".")
-	oidParts := strings.Split(oid, ".")
-	oidBytes := make([]int, len(oidParts))
-
-	// Convert the string OID to an array of integers
-	for i := 0; i < len(oidParts); i++ {
-		oidBytes[i], err = strconv.Atoi(oidParts[i])
+	n := strings.Count(oid, sep) + 1
+	a := make([]int, n)
+	n--
+	i := 0
+	for i < n {
+		m := strings.Index(oid, sep)
+		if m < 0 {
+			break
+		}
+		a[i], err = strconv.Atoi(oid[:m])
 		if err != nil {
 			return nil, fmt.Errorf("unable to parse OID: %w", err)
 		}
+		oid = oid[m+len(sep):]
+		i++
+	}
+	a[i], err = strconv.Atoi(oid)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse OID: %w", err)
 	}
 
-	mOid, err := marshalObjectIdentifier(oidBytes)
+	return a[:i+1], nil
+}
 
+func marshalOID(oid string) ([]byte, error) {
+	var err error
+
```

Comment:
If we are touching this function should we also consider zooming out a bit, and consider the following:
* adding an empty oid string check, and return an error if detected?
* check if OID length >= length as defined in [rfc2578](https://tools.ietf.org/html/rfc2578#section-7.1.3), and return an error if detected?

Reply:
Added validation of sub-identifiers according to clause 3.5 [rfc2578](https://tools.ietf.org/html/rfc2578#section-3.5) , as well as checking for an empty string

Reply:
Like this approach a lot. Thank you